agent/api/api_client_test.go

```diff
@@ -0,0 +1,110 @@
+package api
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/agent/auth"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	svc "github.com/aws/amazon-ecs-agent/agent/ecs_client/ecs_autogenerated_client"
+)
+
+type MockAmazonEC2ContainerServiceV20141113Client struct {
+	svc.AmazonEC2ContainerServiceV20141113Client
+
+	lastRequest interface{}
+}
+
+type TestSubmitContainerStateChangeResponse struct {
+	req svc.SubmitContainerStateChangeRequest
+
+	svc.SubmitContainerStateChangeResponse
+}
+
+func (mock *MockAmazonEC2ContainerServiceV20141113Client) SubmitContainerStateChange(req svc.SubmitContainerStateChangeRequest) (svc.SubmitContainerStateChangeResponse, error) {
+	mock.lastRequest = req
+
+	return svc.NewSubmitContainerStateChangeResponse(), nil
+}
+
+func TestSubmitContainerStateChange(t *testing.T) {
+	client := NewECSClient(auth.TestCredentialProvider{}, &config.Config{Cluster: "default"}, false)
+	mockSvcClient := &MockAmazonEC2ContainerServiceV20141113Client{}
+	client.(*ApiECSClient).serviceClientFn = func() (svc.AmazonEC2ContainerServiceV20141113, error) {
+		return mockSvcClient, nil
+	}
+	err := client.SubmitContainerStateChange(ContainerStateChange{
```

Comment:
I'd prefer that each of these SubmitContainerStateChange invocations was a separate test.